pkg/kafka/config.go

```diff
@@ -141,6 +158,11 @@ func GetConsolidatedConfig(jsonRawConf json.RawMessage, env map[string]string, a
 		// TODO: get rid of envconfig and actually use the env parameter...
 		return result, err
 	}
+
+	if envConfig.AuthMechanism.String != "none" && !envConfig.User.Valid && !envConfig.Password.Valid {
```

Comment:
Shouldn't this be 
```suggestion
	if envConfig.AuthMechanism.String != "none" && (!envConfig.User.Valid || !envConfig.Password.Valid) {
```
? 

Although to be honest I do think the check should be that if one of them is set ... all the others need to be as well :thinking: but that will be a bit ... wordy 

Can you please also add some tests for this :grin:

Reply:
I've tested removing this whole condition to see what happened when one of the parameters was missing. Sarama already validates these parameters and shows more accurate errors, do you think we can stick with those?

e.g. 
```ERRO[0015] could not create the 'kafka' output: kafka: invalid configuration (Net.SASL.User must not be empty when SASL is enabled)```

Reply:
I've added some tests to cover this scenario 😄